lib/platform-stats.js

```diff
@@ -165,29 +284,57 @@ export const mapParticipantsToIds = async (pgClient, participantsSet) => {
     INSERT INTO participants (participant_address)
     SELECT UNNEST($1::TEXT[]) AS participant_address
     ON CONFLICT(participant_address) DO UPDATE
-      -- this no-op update is needed to populate "RETURNING id"
+      -- this no-op update is needed to populate "RETURNING id, participant_address"
       SET participant_address = EXCLUDED.participant_address
-    RETURNING id
+    RETURNING id, participant_address
   `, [
     newAddresses
   ])
 
   assert.strictEqual(created.length, newAddresses.length)
-  return ids.concat(created.map(r => r.id))
+  for (const { id, participant_address: participantAddress } of created) {
+    participantsMap.set(participantAddress, id)
+  }
+
+  return participantsMap
 }
 
 /**
  * @param {import('./typings.js').CreatePgClient} createPgClient
+ * @param {object} options
+ * @param {Array<Function>} [options.functionsToRun]
+ */
+export const refreshDatabase = async (
+  createPgClient,
+  {
+    functionsToRun = [
+      updateTopMeasurementParticipants,
+      aggregateAndCleanUpRecentData,
+      updateMonthlyActiveStationCount
+    ]
+  } = {}
+) => {
+  const pgClient = await createPgClient()
+
+  const results = await Promise.allSettled(
+    functionsToRun.map(func => func(pgClient))
+  )
+  results.forEach((result, index) => {
+    if (result.status === 'rejected') {
+      console.error(`Error running function ${functionsToRun[index].name}:`, result.reason)
+      Sentry.captureException(result.reason)
+    }
+  })
+  await pgClient.end()
+}
+
+/**
+ * @param {import('pg').Client} pgClient
  */
-export const updateTopMeasurementParticipants = async (createPgClient) => {
+export const updateTopMeasurementParticipants = async (pgClient) => {
   try {
     await pRetry(async () => {
-      const pgClient = await createPgClient()
-      try {
-        await pgClient.query('REFRESH MATERIALIZED VIEW top_measurement_participants_yesterday_mv')
-      } finally {
-        await pgClient.end()
-      }
+      await pgClient.query('REFRESH MATERIALIZED VIEW top_measurement_participants_yesterday_mv')
     }, {
       retries: 3,
```

Comment:
I remember the context was because `index.js` [calls this function today directly](https://github.com/filecoin-station/spark-evaluate/blob/ae885a9dcac95a9c3265634511eae5779e7370a5/index.js#L144), and only has access to the `createPgClient` function (i.e. it does not instantiate a pgClient). With these changes, since we are now doing more than just `updateTopMeasurementParticipants` during the scheduled execution, I "bubbled up" the creation of the client to be used for each function.

We could instead pass `createPgClient` to each of the `functionsToRun` in `refreshDatabase()`, and have them each responsible for creating a new client. What would you prefer?

Reply:
> I remember the context was because `index.js` [calls this function today directly](https://github.com/filecoin-station/spark-evaluate/blob/ae885a9dcac95a9c3265634511eae5779e7370a5/index.js#L144), and only has access to the `createPgClient` function (i.e. it does not instantiate a pgClient). > 

Thanks, that makes sense!

> With these changes, since we are now doing more than just `updateTopMeasurementParticipants` during the scheduled execution, I "bubbled up" the creation of the client to be used for each function.

Makes sense 👍🏻 

> We could instead pass `createPgClient` to each of the `functionsToRun` in `refreshDatabase()`, and have them each responsible for creating a new client. What would you prefer?

If we pass `createPgClient` to each function to run, then each function must deal with the complexity of creating a new client and releasing it in a `finally` block. I think this additional complexity is not worth it right now.

Let's keep what you have now in this pull request. 👍🏻